working_villagers/jobs/dyemixer.lua

```diff
@@ -0,0 +1,1489 @@
+	--if iteration == nil then iteration = 0 end -- TODO
+	if self.job_data.wools.dye_names[iteration] ~= nil then
+	for key, value in pairs(self.job_data.wools.dye_names[iteration][ab]) do
+		if item_name==key then
+			return value
+		end
+	end
+	end
+	for key, value in pairs(wools.dye_groups) do
+		if minetest.get_item_group(item_name, key) > 0 then
+			return value;
+		end
+	end
+	return nil
+end
+function wools.is_dye(self, item_name, iteration,ab)
+	assert(item_name ~= nil)
+	assert(iteration ~= nil)
+	assert(ab        ~= nil)
+  local data = wools.get_dye(self, item_name, iteration,ab);
+  return data ~= nil
+end
+
+function wools.get_target(self, item_name, iteration)
+	assert(item_name ~= nil)
+	assert(iteration ~= nil)
+	if self.job_data.wools.target[iteration] == nil then return nil end
+	for key, value in pairs(self.job_data.wools.target[iteration]) do
+		if item_name==key then
+			return value
+		end
+	end
+end
+function wools.is_target(self, item_name, iteration)
+	assert(item_name ~= nil)
+	assert(iteration ~= nil)
+  local data = wools.get_target(self, item_name, iteration);
+  return data ~= nil
+end
+
+
+
+
+
+function dyemixers.get_dyemixer(item_name)
+	assert(item_name ~= nil)
+	for key, value in pairs(dyemixers.names) do
+		if item_name==key then
+			return value
+		end
+	end
+	return nil
+end
+
+function dyemixers.is_dyemixer(item_name)
+	assert(item_name ~= nil)
+	local data = dyemixers.get_dyemixer(item_name);
+	return data ~= nil
+end
+
+local function find_dyemixer_node(pos)
+	assert(pos       ~= nil)
+	local node = minetest.get_node(pos);
+	local data = dyemixers.get_dyemixer(node.name);
+	return data ~= nil
+end
+
+local searching_range = {x = 10, y = 3, z = 10}
+
+local function put_func(villager,stack)
+	assert(villager  ~= nil)
+	assert(stack     ~= nil)
+	local ntarget = villager.job_data.wools.target
+	      ntarget = #ntarget
+	for iteration=0,ntarget,1 do
+		if wools.is_dyingsupplies(villager,stack:get_name(), iteration) then -- TODO all iterations
+			return false
+		end
+	end
+	return true;
+end
+local function take_func(villager,stack)
+	assert(villager  ~= nil)
+	assert(stack     ~= nil)
+	local item_name = stack:get_name()
+	local ntarget = villager.job_data.wools.target
+	      ntarget = #ntarget
+	for iteration=0,ntarget,1 do
+		if wools.is_dyingsupplies(villager,item_name, iteration) then
+			local inv = villager:get_inventory()
+			local itemstack = ItemStack(item_name)
+			itemstack:set_count(wools.get_dyingsupplies(villager,item_name, iteration)) -- TODO all iterations
+			if (not inv:contains_item("main", itemstack)) then
+				return true
+			end
+		end
+	end
+	return false
+end
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+local function take_func2(villager,stack)
+	assert(villager  ~= nil)
+	assert(stack     ~= nil)
+	-- TODO take all dyes and non-target wool
+	local item_name = stack:get_name()
+	local inv = villager:get_inventory()
+	return (inv:room_for_item("main", stack))
+end
+
+
```

Comment:
If there are more than thousand lines in one file, maybe consider making them more modular and moving code to other files, so it is easier to find code dedicated to a certain functionality.

Reply:
yeah that's awful. fixed.